File: 백준/Silver/11060. 점프 점프/점프 점프.js
// 입력을 받아오기 위해 아래 두줄의 코드를 사용하세요. (수정 금지)
const fs = require('fs');
const stdin = fs.readFileSync('/dev/stdin').toString().trim();

let [n, maps] = stdin.split("\n");
n = Number(n)
maps = maps.split(" ").map(Number);
let costs = new Array(n).fill(Infinity);
costs[0] = 0;

let que = [[0, 0]];
while(que.length) {
    let [x, cnt] = que.shift();
    for(let i = 1; i <= maps[x]; i++ ){
        let nx = x + i;
        if(nx < n && cnt + 1 < costs[nx]){
            costs[nx] = cnt + 1;
            que.push([nx, cnt + 1]);
        }
    }
}

console.log(costs.at(-1) == Infinity ? -1 : costs.at(-1));

File: 백준/Silver/11060. 점프 점프/점프 점프.py
import sys
from collections import deque

input = sys.stdin.readline

n = int(input())
maps = list(map(int, input().split()))
costs = [float('inf')] * (n)
costs[0] = 0

que = deque()
que.append((0, 0))

while que:
    x, cnt = que.popleft()

    for i in range(1, maps[x] + 1):
        nx = x + i
        if nx < n and cnt + 1 < costs[nx]:
            costs[nx] = cnt + 1
            que.appendleft((nx, cnt + 1))

print(costs[-1] if costs[-1] != float('inf') else -1)
